libs/ngx-aws-deploy/src/lib/deploy/index.ts

```diff
@@ -95,14 +96,28 @@ export default createBuilder(
         const success = await uploader.upload(files, filesPath);
         if (success) {
           context.logger.info('✔ Finished uploading files...');
-          return { success: true };
+
+          context.logger.info('Start CloudFront invalidation...');
+          const cloudFront = new CloudFront(context, deployConfig);
+          const success = await cloudFront.invalidate();
+          if (success) {
+            context.logger.info('✔ Finished CloudFront invalidation...');
+            return { success: true };
+          } else {
+            context.logger.error(`❌  Error during CloudFront invalidation`);
+            return {
+              error: `❌  Error during CloudFront invalidation`,
+              success: false,
+            };
+          }
```

Comment:
The one thing I think is missing here is that we should check if there is a distribution ID, if not we dont have to invalidate I think? 

Do you agree on this?

Reply:
Thanks for your feedback.

Think we're already safe in that case. My initial PR contains a simple guard in invalidate() that will cause invalidation to be skipped in case the distribution id was not set:
 
`
if (!this._cfDistributionId) {
  this._context.logger.info('⚠️  Skipping invalidation of CloudFront distribution');
      return true;
    }`